### PR TITLE
Defiler Control QoL

### PR DESCRIPTION
**Pull request by Surrealaser**
## About The Pull Request

Emit Gas now activates immediately when selected.

Radial mode support added for Defiler reagent swapping in addition to existing cycling options.

Refactored Defiler reagent selection.

## Why It's Good For The Game

Improves Defiler QoL.

## Changelog
:cl:
qol: Emit Gas now activates immediately when selected.
qol: Radial mode support added for Defiler reagent swapping in addition to existing cycling options.
code: Refactored reagent selection.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm [hunk 1]
```diff
@@ -191,16 +214,32 @@
...
+	var/mob/living/carbon/xenomorph/X = owner
+	for(var/toxin in GLOB.toxin_type_list)
+		var/datum/reagent/R = GLOB.chemical_reagents_list[toxin]
+		if(R.name == toxin_choice)
+			X.selected_reagent = R.type
+			break
+	to_chat(X, "<span class='notice'>We will now use <b>[toxin_choice]</b>.</span>")
 	update_button_icon()
 	return succeed_activate()
+*/
```
Comment (SplinterGP) [inline code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:242]: ?

**Review by SplinterGP — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm (hunk 1 above)
Comment (Surrealaser) [reply code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:242]: Linters doesn't like radials as alternate keybinds, and cycling is preferable as the primary.

Comment (SplinterGP) [reply code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:242]: Linter doenst what? Add it back and just fix the linting issue.

Comment (Surrealaser) [reply code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:242]: You can't fix the linting issue.

Comment (SplinterGP) [reply code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:242]: if you can't fix the linting issue you're doing something wrong, the linter is there to block stuff done wrong.

File: code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm [hunk 2]
```diff
@@ -191,16 +214,32 @@
...
+
 	var/atom/A = X.selected_reagent
-	to_chat(X, "<span class='notice'>We will now slash with <b>[initial(A.name)]</b>.</span>")
+	to_chat(X, "<span class='notice'>We will now use <b>[initial(A.name)]</b>.</span>")
+	update_button_icon()
+	return succeed_activate()
+
+/*
+/datum/action/xeno_action/select_reagent/alternate_keybind_action()
+	var/toxin_choice = show_radial_menu(owner, owner, GLOB.toxin_images_list, radius = 48)
```
Comment (SplinterGP) [inline code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:230]: the radial is why this sleeps and linters cause issue, make another proc with the rest and just INVOKE_ASYNC that proc in this proc

**Review by SplinterGP — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm [hunk 3]
```diff
@@ -191,17 +214,35 @@
...
-	to_chat(X, "<span class='notice'>We will now slash with <b>[initial(A.name)]</b>.</span>")
+	to_chat(X, "<span class='notice'>We will now use <b>[initial(A.name)]</b>.</span>")
 	update_button_icon()
 	return succeed_activate()
 
+/datum/action/xeno_action/select_reagent/alternate_keybind_action()
+	INVOKE_ASYNC(src, .proc/alternate_keybind_action_async)
+
+/datum/action/xeno_action/select_reagent/proc/alternate_keybind_action_async()
+	var/toxin_choice = show_radial_menu(owner, owner, GLOB.toxin_images_list, radius = 48)
```
Comment (SplinterGP) [inline code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:232]: better proc name would be good, like "select_reagent_radial"

File: code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm [hunk 4]
```diff
@@ -1,3 +1,20 @@
...
+
+//List of huggie types
+GLOBAL_LIST_INIT(toxin_type_list, list(
+		/datum/reagent/toxin/xeno_neurotoxin,
+		/datum/reagent/toxin/xeno_hemodile,
+		/datum/reagent/toxin/xeno_transvitox,
+		))
+
+//List of huggie images
+GLOBAL_LIST_INIT(toxin_images_list,  list(
```
Comment (SplinterGP) [inline code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:13]: need better names, defiler_toxins or something

File: code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm [hunk 5]
```diff
@@ -1,3 +1,20 @@
...
+		DEFILER_NEUROTOXIN = image('icons/mob/actions.dmi', icon_state = DEFILER_NEUROTOXIN),
+		DEFILER_HEMODILE = image('icons/mob/actions.dmi', icon_state = DEFILER_HEMODILE),
+		DEFILER_TRANSVITOX = image('icons/mob/actions.dmi', icon_state = DEFILER_TRANSVITOX),
+		))
```
Comment (SplinterGP) [inline code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:17]: these can probably just be a static var on the proc

**Review by SplinterGP — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm [hunk 6]
```diff
@@ -1,3 +1,13 @@
+#define DEFILER_NEUROTOXIN "Neurotoxin"
+#define DEFILER_HEMODILE "Hemodile"
+#define DEFILER_TRANSVITOX "Transvitox"
```
Comment (SplinterGP) [inline code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:3]: undefine at end of file

**Review by SplinterGP — CHANGES_REQUESTED** [review]

**Review by SplinterGP — APPROVED** [review]

**Pull request merged**